src/main/java/com/github/checkstyle/regression/Main.java

```diff
@@ -34,6 +68,91 @@ private Main() {
      * @throws Exception execute failure
      */
     public static void main(String[] args) throws Exception {
-        // empty for now
+        final Options options = createOptions();
+        final CommandLineParser parser = new DefaultParser();
+        final HelpFormatter formatter = new HelpFormatter();
+        final CommandLine cmd;
+
+        try {
+            cmd = parser.parse(options, args);
+            final String repoPath = cmd.getOptionValue(OPT_REPO);
+            final String branch = cmd.getOptionValue(OPT_BRANCH);
+            final boolean stopAfterGenerateConfig = cmd.hasOption(OPT_STOP_AFTER_GENERATE_CONFIG);
+
+            final File config = generateConfig(repoPath, branch);
+            System.out.println("config generated at " + config.getAbsolutePath());
+            if (!stopAfterGenerateConfig) {
+                final String testerPath = cmd.getOptionValue(OPT_TESTER);
+                final File report = ReportGenerator.generate(testerPath, repoPath, branch, config);
+                System.out.println("report generated at " + report.getAbsolutePath());
+            }
+        }
+        catch (ParseException ex) {
```

Comment:
> 1. wrong format 2.missing arguments 3. wrong argument number etc.

Than why is it covering everything if it is only for user input? What lines are throwing this?